src/trg-client.c

```diff
@@ -240,10 +241,43 @@ TrgPrefs *trg_client_get_prefs(TrgClient * tc)
     return tc->priv->prefs;
 }
 
+static void trg_client_insert_header (JsonNode *header_node, GList **output)
+{
+  const gchar *key = NULL, *value;
+    JsonObject *header_object;
+    if (!header_node)
+        return;
+
+    header_object = json_node_get_object(header_node);
+    if (!header_object)
+      return;
+
+    key = json_object_get_string_member(header_object, TRG_PREFS_KEY_CUSTOM_HEADER_NAME);
+    value = json_object_get_string_member(header_object, TRG_PREFS_KEY_CUSTOM_HEADER_VALUE);
+    if (key && value)
+      *output = g_list_prepend(*output, g_strdup_printf("%s: %s", key, value));
+}
+
+static GList *trg_client_headers_array_to_list(JsonArray *array)
+{
+    GList *output_headers = NULL;
+    GList *nodes;
+    if (!array)
+        return NULL;
+
+    nodes = json_array_get_elements(array);
+    if (!nodes)
+      return NULL;
+
+    g_list_foreach(nodes, (GFunc)trg_client_insert_header, &output_headers);
```

Comment:
Personal preference but I think a regular `for` loop is far more readable here, less dealing with passing around and modifying pointers.

Reply:
i thought this would be more readable and makes having smaller functions but i can also change it, do you want me to do it?

Reply:
Yes please.